nwbwidgets/base.py

```diff
@@ -213,12 +212,15 @@ def plot_traces(time_series: TimeSeries, time_start=0, time_duration=None, trace
         Index range of traces to view
     title: str, optional
     ylabel: str, optional
-
     Returns
     -------
-
     """
-
+    
+    if type(time_series.data) != np.ndarray:
+        time_series.data = np.asarray(time_series.data)
+    if len(time_series.data.shape) == 1:
+        time_series.data = time_series.data.reshape((-1,1))
+    
```

Comment:
Good ideas. However, both of these lines are going to be problematic, because they both require reading the data. When you write an NWB file and then read it back, it presents datasets not as lists or np.arrays, but has `h5py.Dataset`s. The data is not read, but the `h5py.Dataset`s provide "views" into the data. The data is only read when you try to slice it (e.g. `time_series.data[:5]`). This is important because the data is sometimes too big to fit into memory, so we only want to read data that we are planning to view. 

Also, I would rather not change an attribute of `time_series.data`. In python, that change will persist in that object, even outside the scope of the function. Instead, we should declare a variable that is local to the function, e.g. `data = time_series.data` and use that.

Reply:
Maybe a better approach is to declare a `n_traces` variable instead of using `time_series.data.shape[1]`.

Reply:
as you might assume from the name of the function (plot_traces), I wasn't really planning on visualizing a single trace, but yeah I think it is better if this function does not break in that case.

Reply:
@Armin12  please address this